lib/s3WriteStream.js

```diff
@@ -153,29 +153,36 @@
         });
     };
 
-    MultiPartManager.prototype.complete = function () {
-        var self = this;
-        return this.partNumber ? this.uploadId().then(function (uploadId) {
-            self.flush();
-            return Promise.all(self.parts).then(function (parts) {
-                return new Promise(function (resolve, reject) {
-                    self.client.completeMultipartUpload({
-                        Bucket: self.bucket,
-                        Key: self.key,
-                        UploadId: uploadId,
-                        MultipartUpload: {
-                            Parts: parts
-                        }
-                    }, function (err, data) {
-                        if (err) {
-                            return reject(err);
-                        }
-                        resolve(data);
-                    });
-                });
-            });
-        }) : this.put(); //if we did not reach the part limit of 5M just use putObject
-    };
+  MultiPartManager.prototype.complete = function () {
+    var self = this;
+    return this.partNumber ? this.uploadId().then(function (uploadId) {
+      self.flush();
+      return Promise.all(self.parts).then(function (parts) {
+        // Removing "ServerSideEncryption":"AES256" to resolve issue with S3 change 07/02/23 - AD
+        const partsWithSSERemoved = parts.map(part => {
+          return {
```

Comment:
Won't this code actually remove all other keys? (not just ServerSideEncryption)
In the aws docs it looks like there may be other keys at times (eg. checksums), so should we just explicitly remove only the ServerSideEncryption key from the part if it exists?